src/app/users/get-user.ts

```
@@ -1,65 +1,99 @@
-          (ipObject: IPType) => ipObject.ip === ip,
-        )
-        if (lastIP) {
-          lastIP.lastConnection = user.lastConnection
-        } else if (ipConfig.proxyCheckingEnabled) {
+
+      const lastConnection = new Date()
+
+      const userIp = await usersIp.findById(userId)
+
+      if (userIp instanceof RepositoryError) throw userIp
+
+      if (!ip || !ipConfig.ipRecordingEnabled) {
+        const result = await usersIp.update(userId, lastConnection)
+
+        if (result instanceof Error) {
+          logger.error(
+            { result, userId, ip },
+            "impossible to update user last connection",
+          )
+        }
+
+        return
+      }
+
+      const lastIP = userIp.lastIPs.find((ipObject) => ipObject.ip === ip)
+
+      if (lastIP) {
+        lastIP.lastConnection = lastConnection
+      } else {
+        let ipInfo = {
+          ip,
+          firstConnection: lastConnection,
+          lastConnection: lastConnection,
+        } as IPType
+
+        if (ipConfig.proxyCheckingEnabled) {
           const ipFetcher = IpFetcher()
-          const ipInfo = await ipFetcher.fetchIPInfo(ip as IpAddress)
-          if (!(ipInfo instanceof Error)) {
-            user.lastIPs.push({
-              ip,
-              ...ipInfo,
-              Type: ipInfo.type,
-              firstConnection: user.lastConnection,
-              lastConnection: user.lastConnection,
-            })
+          const ipFetcherInfo = await ipFetcher.fetchIPInfo(ip as IpAddress)
+
+          if (!(ipFetcherInfo instanceof Error)) {
+            ipInfo = { ...ipInfo, ...ipFetcherInfo }
           }
         }
+        userIp.lastIPs.push(ipInfo)
       }
+      const result = await usersIp.update(userId, lastConnection, userIp.lastIPs)
 
-      const updateResult = await users.update(user)
-
-      if (updateResult instanceof RepositoryError) {
-        return updateResult
+      if (result instanceof Error) {
```

Again don't eat the error. Always return it - it will automatically be reported via OpenTelemetry.